src/test/java/ay2021s1_cs2103_w16_3/finesse/testutil/TypicalTransactions.java

```diff
@@ -60,11 +62,29 @@ public static FinanceTracker getTypicalFinanceTracker() {
         FinanceTracker ft = new FinanceTracker();
         for (Transaction transaction : getTypicalTransactions()) {
             ft.addTransaction(transaction);
+            ft.addExpense(new TransactionBuilder(transaction).buildExpense());
+            ft.addIncome(new TransactionBuilder(transaction).buildIncome());
         }
         return ft;
     }
 
     public static List<Transaction> getTypicalTransactions() {
         return new ArrayList<>(Arrays.asList(ALICE, BENSON, CARL, DANIEL, ELLE, FIONA, GEORGE));
     }
+
+    public static List<Expense> getTypicalExpenses() {
+        List<Expense> typicalExpenses = new ArrayList<>();
+        for (Transaction transaction : getTypicalTransactions()) {
+            typicalExpenses.add(new TransactionBuilder(transaction).buildExpense());
+        }
+        return typicalExpenses;
+    }
+
+    public static List<Income> getTypicalIncomes() {
+        List<Income> typicalIncomes = new ArrayList<>();
+        for (Transaction transaction : getTypicalTransactions()) {
+            typicalIncomes.add(new TransactionBuilder(transaction).buildIncome());
+        }
+        return typicalIncomes;
+    }
```

Comment:
```suggestion
    public static List<Expense> getTypicalExpenses() {
        return getTypicalTransactions().stream()
                .map(TransactionBuilder::new)
                .map(TransactionBuilder::buildExpense)
                .collect(Collectors.toList());
    }

    public static List<Income> getTypicalIncomes() {
        return getTypicalTransactions().stream()
                .map(TransactionBuilder::new)
                .map(TransactionBuilder::buildIncome)
                .collect(Collectors.toList());
    }
```

Might be possible to further simplify by extracting out the common parts.